### PR TITLE
BUILD and WORKSPACE fixes for bazel 0.6.1 

### LINKED ISSUE #1365: build broken with bazel 0.6.0
I've gotten build errors as follows:

```
% bazel build //...
ERROR: /usr/local/google/home/mukai/.cache/bazel/_bazel_mukai/23d88ec2f9449b2c9d846b080b7cc444/external/io_bazel_rules_go/go/private/binary.bzl:226:48: name 'lib' is not defined (did you mean 'libs'?)
ERROR: error loading package '': Extension 'go/private/binary.bzl' has errors
ERROR: error loading package '': Extension 'go/private/binary.bzl' has errors
INFO: Elapsed time: 0.132s
FAILED: Build did NOT complete successfully (0 packages loaded)
```

Seems this is because of bazel 0.6.0?

I had another environment which still stays on bazel 0.5.4. Succeeded to build, but started reporting the same error after upgraded to 0.6.0.

### PR DESCRIPTION
**Pull request by lizan**
- Update rules_go to latest
- Cleaning up deps from WORKSPACE which are built-in in rules_go
- Gazelify all BUILD files
- Fix test/repositories.bzl for proxy build

Based on #1366, fixes #1365.

**Release note**:

```release-note
None
```

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/1423)

### CODE REVIEW

**Comment by jmuk** [comment]
This looks good, but I'm a bit worried about having bunch of "importpath=" line in every build target.

Actually mixer builds successfully without them, but gazelle can't tell if they are really necessary or not and it adds these lines regardless the necessity (there's a TODO on gazelle side, see https://github.com/bazelbuild/rules_go/blob/master/go/tools/gazelle/rules/generator.go#L155). Considering this situation, my PR added a nasty sed line to exclude them.

I'm not sure which is appealing to other mixer devs, I want to follow others.

**Comment by jmuk** [comment]
ah, by the way, this implicitly upgrades go version to 1.9.1, right? We haven't discussed this much -- personally I'm totally fine with the upgrade since it builds successfully, but reviewers would need to be aware of it (go_version can be specified in `go_register_toolchain` if we really don't want the upgrade).

**Comment by kyessenov** [comment]
Yes, @lizan please pin go version to 1.8.3 so we can explicitly pick a version. rules_go has all the toolchains available. 

I don't mind adding the silly sed to strip importpath to minimize the impact. I think gazelle is likely to remove that in some future form.

File: adapter_author_deps.bzl [hunk 1]
```diff
@@ -4,24 +4,18 @@ def mixer_adapter_repositories():
 
     native.git_repository(
         name = "org_pubref_rules_protobuf",
-        commit = "eafd42ce6471ce3ea265729c85e18e6180dea620",  # Sept 22, 2017 (genfiles path calculation fix)
+        commit = "ff3b7e7963daa7cb3b42f8936bc11eda4b960926",  # Sept 22, 2017 (genfiles path calculation fix)
```
Comment (kyessenov) [inline adapter_author_deps.bzl]: update the date?

**Comment by lizan** [comment]
Sure, golang upgrade shouldn't be implied by this PR, pinned to 1.8.3. Also revert BUILD files and re-gazellified with `importpath` sed.

**Comment by jmuk** [comment]
@lizan, maybe you need to update io_bazel_rules_docker dependency too for the presubmit failure.

**Comment by jmuk** [comment]
/lgtm

The smoketest failure seems happening because `istio/istio` repository still refers to the old rules_go...

**Comment by mandarjog** [comment]
@lizan please rebase the PR.
It will force you to change allowable bazel versions.
0.5.3 - 0.6.1 seems to work.

**Pull request merged**